tracker/utils.py

```diff
@@ -317,3 +322,40 @@ def attach_link_to_issue(issue_title: str, issue_link: str) -> str:
     title = f'<a href="{issue_link}">{issue_title}</a>'
     return title
 
+
+def get_time_before_deadline(issue: dict) -> str:
+    """
+    Returns the time remaining before the deadline of an assigned issue.
+    If the issue has no assignee or deadline, returns appropriate messages.
+
+    :param issue: The issue dictionary containing information about the issue.
+    :return: Time remaining in a human-readable format.
+    """
+
+    assignment_info = check_issue_assignment_events(issue)
+    assigned_at = assignment_info.get("assigned_at")
+
+    if not assigned_at:
+        return "This issue is not assigned."
+
+    deadline_str = issue.get("due_on")
```

Comment:
No, the original GH issue object doesn't have this field. You should calculate it yourself.

Here you have `time_limit` filed (in hours).
https://github.com/KostCode-Organization/hackathon_bot/blob/main/tracker/models.py#L158

You should just check what repository the issue if related to -> get the corresponding `Repository` object and finally calculate the time till deadline based on `time_limit` value